airflow/runtimes/container_runtime.go

```diff
@@ -48,9 +50,9 @@ func GetContainerRuntime() (ContainerRuntime, error) {
 	// Return the appropriate container runtime based on the binary discovered.
 	switch containerRuntime {
 	case docker:
-		return CreateDockerRuntime(new(dockerEngine), new(osChecker)), nil
+		return CreateDockerRuntimeWithDefaults(), nil
 	case podman:
-		return CreatePodmanRuntime(new(podmanEngine), new(osChecker)), nil
+		return CreatePodmanRuntimeWithDefaults(), nil
```

Comment:
Instead of specifying these concrete implementations out here, we've added new constructor functions to construct with the default concrete implementations. This lets us encapsulate the new logic to determine the specific `Engine` to plug in.